auth.php

```diff
@@ -21,1148 +21,17 @@
-    }
-
-    public static function saml2_authproc_filters_hook() {
-        $authprocfilters = [];
-        $authprocfilters[50] = array(
-            'class' => 'core:AttributeMap',
-            'oid2name',
-        );
-        $callbacks = get_plugins_with_function('extend_auth_saml2_proc', 'lib.php');
-        foreach ($callbacks as $plugins) {
-            foreach ($plugins as $pluginfunction) {
-                $filters = $pluginfunction();
-                foreach ($filters as $key => $value) {
-                    $key = self::check_filters_priority($key, $authprocfilters);
-                    $authprocfilters[$key] = $value;
-                }
-            }
-        }
-        return $authprocfilters;
-    }
-
-    public static function check_filters_priority($priority, $filters) {
-        $uniquekey = false;
-        while (!$uniquekey) {
-            if (!array_key_exists($priority, $filters)) {
-                $uniquekey = true;
-            } else {
-                $priority++;
-            }
-        }
-        return $priority;
-    }
-
-    public function get_saml2_directory() {
-        global $CFG;
-        $directory = "{$CFG->dataroot}/saml2";
-        if (!file_exists($directory)) {
-            mkdir($directory);
-        }
-        return $directory;
-    }
-
-    public function get_file($file) {
-        return $this->get_saml2_directory() . '/' . $file;
-    }
-
-    public function get_file_sp_metadata_file() {
-        return $this->get_file($this->spname . '.xml');
-    }
-
-    /**
-     * @param string|array $url The string with the URL or an array with all URLs as keys.
-     * @return string Metadata file path.
-     */
-    public function get_file_idp_metadata_file($url) {
-        if (is_object($url)) {
-            $url = (array)$url;
-        }
-
-        if (is_array($url)) {
-            $url = array_keys($url);
-            $url = implode("\n", $url);
-        }
-
-        $filename = md5($url) . '.idp.xml';
-        return $this->get_file($filename);
-    }
-
-    /**
-     * A debug function, dumps to the php log
-     *
-     * @param string $msg Log message
-     */
-    private function log($msg) {
-        if ($this->config->debug) {
-            // @codingStandardsIgnoreStart
-            error_log('auth_saml2: ' . $msg);
-            // @codingStandardsIgnoreEnd
-
-            // If SSP logs to tmp file we want these to also go there.
-            if ($this->config->logtofile) {
-                require_once('setup.php');
-                SimpleSAML\Logger::debug('auth_saml2: ' . $msg);
-            }
-        }
-    }
-
-    /**
-     * Returns a list of potential IdPs that this authentication plugin supports.
-     * This is used to provide links on the login page.
-     *
-     * @param string $wantsurl the relative url fragment the user wants to get to.
-     *
-     * @return array of IdP's
-     */
-    public function loginpage_idp_list($wantsurl) {
```

Comment:
This function cannot be moved its a callback that core calls

Reply:
Actually no I see its moved to parent class, but is there a good reason this and the other sibling methods that are called by core should be in the parent rather than in this one? ie should is_internal, is_configured, loginpage_hook etc stay here?

Reply:
Just to be clear, the core convention auth.php located at root now looks like:
```
class auth_plugin_saml2 extends \auth_saml2\auth {
}
```
So, all methods that core is expecting to be there are reachable. We have got core `auth_oauth2` plugin that uses the same appraoch, it is just more convenient. If at any other place the plugin is referenced by child, it will not fail either (e.g. in unittest it its instance is requested as `get_auth_plugin('saml2')` and all works).